itests/kit/node_unmanaged.go

```diff
@@ -93,271 +128,486 @@ func NewTestUnmanagedMiner(t *testing.T, full *TestFullNode, actorAddr address.A
+
+	si, err := tm.getCommittedSector(sectorNumber)
+	req.NoError(err)
+
+	updateProofType := abi.SealProofInfos[si.proofType].UpdateProof
+	var pieces []abi.PieceInfo
+	var snapProof []byte
+	var newSealedCid, newUnsealedCid cid.Cid
+
+	if !tm.mockProofs {
+		var unsealedPath string
+		var err error
+		pieces, unsealedPath, err = tm.mkStagedFileWithPieces(si.proofType)
+		req.NoError(err)
+
+		s, err := os.Stat(si.sealedSectorPath)
+		req.NoError(err)
+
+		randomBytes := make([]byte, s.Size())
+		_, err = io.ReadFull(rand.Reader, randomBytes)
+		req.NoError(err)
+
+		updatePath, err := mkTempFile(tm.t, bytes.NewReader(randomBytes), uint64(s.Size()))
+		req.NoError(err)
+		req.NoError(updatePath.Close())
+		updateDir := filepath.Join(tm.t.TempDir(), fmt.Sprintf("update-%d", sectorNumber))
+		req.NoError(os.MkdirAll(updateDir, 0700))
+
+		newSealedCid, newUnsealedCid, err = ffi.SectorUpdate.EncodeInto(updateProofType, updatePath.Name(), updateDir,
+			si.sealedSectorPath, si.cacheDirPath, unsealedPath, pieces)
+		req.NoError(err)
+
+		vp, err := ffi.SectorUpdate.GenerateUpdateVanillaProofs(updateProofType, si.sealedCid,
+			newSealedCid, newUnsealedCid, updatePath.Name(), updateDir, si.sealedSectorPath, si.cacheDirPath)
+		req.NoError(err)
+
+		snapProof, err = ffi.SectorUpdate.GenerateUpdateProofWithVanilla(updateProofType, si.sealedCid, newSealedCid, newUnsealedCid, vp)
+		req.NoError(err)
+	} else {
+		pieces = []abi.PieceInfo{{
+			Size:     abi.PaddedPieceSize(tm.options.sectorSize),
+			PieceCID: cid.MustParse("baga6ea4seaqlhznlutptgfwhffupyer6txswamerq5fc2jlwf2lys2mm5jtiaeq"),
+		}}
+		snapProof = []byte{0xde, 0xad, 0xbe, 0xef}
+		newSealedCid = cid.MustParse("bagboea4b5abcatlxechwbp7kjpjguna6r6q7ejrhe6mdp3lf34pmswn27pkkieka")
+	}
+
+	tm.waitForMutableDeadline(sectorNumber)
+
+	var manifest []miner14.PieceActivationManifest
+	for _, piece := range pieces {
+		manifest = append(manifest, miner14.PieceActivationManifest{
+			CID:  piece.PieceCID,
+			Size: piece.Size,
+		})
+	}
+
+	head, err := tm.FullNode.ChainHead(tm.ctx)
+	req.NoError(err)
+
+	sl, err := tm.FullNode.StateSectorPartition(tm.ctx, tm.ActorAddr, sectorNumber, head.Key())
+	req.NoError(err)
+
+	tm.log("Submitting ProveReplicaUpdates3 for sector %d ...", sectorNumber)
+
+	params := &miner14.ProveReplicaUpdates3Params{
+		SectorUpdates: []miner14.SectorUpdateManifest{
+			{
+				Sector:       sectorNumber,
+				Deadline:     sl.Deadline,
+				Partition:    sl.Partition,
+				NewSealedCID: newSealedCid,
+				Pieces:       manifest,
+			},
+		},
+		SectorProofs:               [][]byte{snapProof},
+		UpdateProofsType:           updateProofType,
+		RequireActivationSuccess:   true,
+		RequireNotificationSuccess: false,
+	}
+	r, err := tm.SubmitMessage(params, 1, builtin.MethodsMiner.ProveReplicaUpdates3)
+	req.NoError(err)
+	req.True(r.Receipt.ExitCode.IsSuccess())
+
+	si.pieces = pieces
+	si.sealedCid = newSealedCid
+	si.unsealedCid = newUnsealedCid
+	tm.setCommittedSector(si)
```

Comment:
Right so until this is done, WindowPost loop will use the old `sectorInfo` and will start using the new one as soon as this is in.